shop/src/components/payment/crypto/CryptoDirect.js

```diff
@@ -92,22 +95,46 @@ const PayWithCryptoDirect = ({ submit, encryptedData, onChange, loading }) => {
 
   const acceptedTokens = config.acceptedTokens
 
+  useEffect(() => {
+    onChange(
+      (() => {
+        if (wallet.signer) {
+          wallet.signer
+            .getBalance() //returns a promise that, if resolved, will give the balance of the wallet in wei (BigNumber)
```

Comment:
Don't think this will work either. The signer won't be set in the Alchemy providers we use on prod. It's better to get balance of a wallet this way: `await provider.getBalance(wallet.address)`.   

You can get the `provider` from `useWallet` hook